dll/cpl/sysdm/environment.c

```diff
@@ -17,6 +23,148 @@ typedef struct _VARIABLE_DATA
     LPTSTR lpCookedValue;
 } VARIABLE_DATA, *PVARIABLE_DATA;
 
+typedef struct _ENVIRONMENT_EDIT_DIALOG_DATA
+{
+    BOOL bIsFriendlyUI;
```

Comment:
This seems to be redundant with dwDlgID.
Indeed, when `dwDlgID == IDD_EDIT_VARIABLE`, we know it's not the friendly UI, but when `dwDlgID == IDD_EDIT_VARIABLE_FANCY`, we know it's the friendly UI.